Cookbook/Proposals/SmarterDiffs.md

```diff
@@ -0,0 +1,97 @@
+    makeState(status: .idle),
+    makeState(status: .loading),
+    makeState(
+        shouldShowNHSBanner: false, // this value causes failure
+        shouldShowHealthcheckBanner: true,
+        shouldShowTestAndKitsBanner: true,
+        shouldShowGPAtHandStatusCard: true,
+        status: .loaded
+    )
+]))
+```
+
+This will result in diff
+
+```
+Found difference for child some:
+child :
+	shouldShowNHSBanner received: "true" expected: "false"
+```
+
+instead of
+
+```
+expected to equal <[State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.idle, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loading, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: true, shouldShowTestAndKitsBanner: true, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loaded, isInsuranceValid: true)]>, got <[State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.idle, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loading, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: true, shouldShowHealthcheckBanner: true, shouldShowTestAndKitsBanner: true, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loaded, isInsuranceValid: true)]>
+```
+
+As can be seen from the diff it does not show in what exact array item the failure happened. This is something that we will need to adjust along with [this issue](https://github.com/krzysztofzablocki/Difference/issues/1). 
+
+Proposed way to integrate this feature is to copy the implementation (with proper author attribution) and adjust it to our needs, optionally we can make pull requests to the original library.
```

Comment:
Having a local copy might make it a bit easier to work on fixing the issues with library. Once we're confident in our fixes, we could submit PR to original library. Not sure how much of an overhead it would be to maintain our fork.